_posts/2025-08-16-accessible-error-messages-in-forms.md

```diff
@@ -0,0 +1,158 @@
+
+Checking during typing can be confusing. Imagine typing an email address and seeing “Invalid email address” after the first letter. Of course, it’s invalid — you haven’t finished yet! 
+
+That leaves two options: check after leaving a field, or after submission. These can be combined. For certain fields — like dates that must be in the future — immediate checking is helpful. But checking after submission is always necessary.
+
+{: .warning }
+HTML5 form validation is not accessible at this moment. The W3C summarizes the options in [Validating Input ](https://www.w3.org/WAI/tutorials/forms/validation/). "If your web browser supports HTML5, it will not allow you to submit the form without entering text into the input field. Instead, it will display a message that is generated by the web browser itself".  Write custom error messages and provide server side generated error messages. The W3C explains the options in [Validating Input ](https://www.w3.org/WAI/tutorials/forms/validation/).
+
+## How do I indicate that an answer is incomplete or filled out incorrectly?
+
+Use more than colour alone to indicate errors. A user with visual impairments or colour blindness may not notice a red outline.  Colour is helpful — but always include an error message in text too.
+
+* Clearly state whether a field is required.
+* Show errors with more than just colour or an icon.
+* Write out the error in text above the form field.
+* Add a summary of all errors above the form  helps the user quickly see what needs to be corrected.
+
+
+## What makes a good error message?
+
+Messages like “This field is required” or “Invalid value” don’t offer much help. Write clear error messages that explain what’s missing or needs changing.
+
+{: .example }
+“Enter your first name”  
+“Your chosen password is too short. It must be at least 8 characters long.”  
+“The expiration date must be in the future.”
+
+This is far more helpful than a vague generic message.
+
+
+## How do I confirm a form was successfully submitted?
+
+You click "Send" and… nothing happens. Or you're redirected to the homepage. Was the form actually sent? How to know?
+
+Give users clear confirmation that their form has been submitted and what will happen next. You can also include this messsage in a confirmation email.
+
+{: .example }
+A confirmation text could be:  
+Thank you for your registration for our workshop "Knitting socks". A confirmation email has been sent to user@example.com with the time and location of the workshop. If you haven't received an email? [Please contact us](#).
+
+## How can I offer help if a user sets stuck?
+
+If a user gets stuck filling out a complex form, they should be able to easily request help.
+
+Don’t hide contact details at the very bottom of the form — place a short sentence near the top. The footer is also a good place to additionally include contact info. Always offer multiple ways to get in touch on your contact page — not everyone can make a phone call.
```

Comment:
"Don't hide contact details. They shouldn't be buried at the bottom of the form, or isolated on just one page of your site. Include a short sentence at the top of your form, and place contact information in predictable places in your site design. The footer is a common place to include additional contact information. "

Last sentence can stay as is.